app/controllers/carts_controller.rb

```diff
@@ -2,18 +2,30 @@ class CartsController < ApplicationController
 
   def add
     dish = Dish.find(params[:id])
-    if session[:cart_id]
-      show
-    else
-      @cart = Cart.create
-      session[:cart_id] = @cart.id
-    end
+    find_cart
     @cart.add(dish, dish.price)
     flash[:notice] = "#{dish.name} added to cart"
     redirect_to restaurant_path(params[:restaurant_id])
   end
 
+  def remove_item
+    dish = Dish.find(params[:dish_id])
+    find_cart
```

Comment:
Just one thought - will it create a cart for the user, if it doesn't have a cart when it tries to remove a dish?

Reply:
yes it will. I could stop that behavior just not sure if it is needed.

Reply:
Since you'll only show the button if the dish is in the cart, it won't be needed! :)

Reply:
This as well. Before action